retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaRetrofitBuilder.java

```diff
@@ -0,0 +1,221 @@
+ *             .build(HttpClient.class))
+ *     .build();
+ * }
+ * </pre>
+ */
+public final class ArmeriaRetrofitBuilder {
+
+    private static final Pattern GROUP_PREFIX_PATTERN = Pattern.compile("^[_0-9a-z]+$");
+    private static final String DEFAULT_GROUP_PREFIX = "__group__";
+    private static final Function<String, HttpClient> DEFAULT_NEW_CLIENT_FUNCTION =
+            url -> Clients.newClient(ClientFactory.DEFAULT, url, HttpClient.class);
+    public static final String SLASH = "/";
+
+    private final Retrofit.Builder retrofitBuilder;
+    private String baseUrl;
+    private HttpClient baseHttpClient;
+    private String basePath;
+    private String groupPrefix = DEFAULT_GROUP_PREFIX;
+    private Function<String, HttpClient> newClientFunction = DEFAULT_NEW_CLIENT_FUNCTION;
+
+    /**
+     * Creates a {@link Retrofit.Builder} with {@link ArmeriaCallFactory} using the specified
+     * url.
+     *
+     * @see Builder#baseUrl(String)
+     */
+    public ArmeriaRetrofitBuilder(String baseUrl) {
+        String path = URI.create(baseUrl).getPath();
+        if (!path.isEmpty() && !SLASH.equals(path.substring(path.length() - 1))) {
+            throw new IllegalArgumentException("baseUrl must end in /: " + baseUrl);
+        }
+        this.baseUrl = baseUrl;
+        retrofitBuilder = new Retrofit.Builder();
+    }
+
+    /**
+     * Creates a {@link Retrofit.Builder} with {@link ArmeriaCallFactory} using the specified
+     * {@link HttpClient} instance.
+     */
+    public ArmeriaRetrofitBuilder(HttpClient baseHttpClient) {
+        this(baseHttpClient, SLASH);
+    }
+
+    /**
+     * Creates a {@link Retrofit.Builder} with {@link ArmeriaCallFactory} using the specified
+     * {@link HttpClient} instance.
+     */
+    public ArmeriaRetrofitBuilder(HttpClient baseHttpClient, String basePath) {
+        final String path = baseHttpClient.uri().getPath();
+        if (!path.isEmpty() && !SLASH.equals(path)) {
+            throw new IllegalArgumentException(
+                    "ArmeriaRetrofitBuilder doesn't support http client's uri contains any path element," +
+                    " current path: " + path +
+                    ". If you want to using uri with path, please using constructor with basePath argument.");
+        }
+        if (!basePath.isEmpty()) {
+            if (!SLASH.equals(basePath.substring(0, 1))) {
+                throw new IllegalArgumentException("basePath must begin with /: " + basePath);
+            }
+            if (!SLASH.equals(basePath.substring(basePath.length() - 1))) {
+                throw new IllegalArgumentException("basePath must end in /: " + basePath);
+            }
+        }
+        this.baseHttpClient = baseHttpClient;
+        this.basePath = basePath;
+        retrofitBuilder = new Retrofit.Builder();
+    }
+
+    /**
+     * Set different group name prefix for avoiding conflict of endpoint which starts with __group__.
```

Comment:
yeah, and I just notice that hostname not allowed underscore, so using `group_` and remove unnecessary prefix configuration.